frontend/src/main/scala/bloop/bsp/BloopBspServices.scala

```diff
@@ -662,6 +668,56 @@ final class BloopBspServices(
     }
   }
 
+  def jvmTestEnvironment(
+      params: bsp.JvmEnvironmentParams
+  ): BspEndpointResponse[bsp.JvmEnvironmentResult] = {
+    ifInitialized(None) { (state: State, logger: BspServerLogger) =>
+      mapToProjects(params.targets, state) match {
+        case Left(error) =>
+          logger.error(error)
+          Task.now((state, Left(JsonRpcResponse.invalidRequest(error))))
+
+        case Right(projects) =>
+          val environmentEntries: List[Either[String, JvmEnvironmentEntry]] = for {
+            (id, project) <- projects.toList
+            dag = state.build.getDagFor(project)
+            fullClasspath = project.fullClasspath(dag, state.client).map(_.toString)
```

Comment:
We should update the implementation of `project.fullClasspath` to automatically use manifest jars. It would be good to move the logic from here into some reusable method, or change `JvmProcessForker` to use `project.fullClasspath`

https://github.com/scalacenter/bloop/blob/dfd03f5333c2fc292db78bfcd4205a0cdd828967/frontend/src/main/scala/bloop/exec/JvmProcessForker.scala#L158-L189

I would personally prefer if the manifest jar did not get deleted after execution finishes, I've had several cases where I have been unable to debug the java invocation because the manifest jar got deleted.

Reply:
This will change also fix an issue where `bloop console ...` currently fails for large classpath since I see `bloop console` also uses `project.fullClasspath`

Reply:
I'm wondering what would be the right approach there. Currently we create `manifest.jar` as a tmp file, so keeping it forever would lead to a file leak. We have a few choices here:

1. Let it leak
2. Change file location from a random tmp file to a deterministic one
3. Generate manifest.jar on intellij site while running the app

Reply:
I would be fine with 1 or 2. The manifest jars are normally small since they only contain a MANIFEST.MD text file that points to other paths on disk so I don't think it's a big concern if we use 1) with temporary files and let them leak. Option 2 might be good if we use a stable temporary directory and regularly garbage collect (for example when the directory contains more than N number of manifest jars). I would like to avoid 3 since it means that every client of `jvmTestEnvironment` will need to deal with manifest jars.

What do you think @jvican ?

Reply:
I think it's helpful that the manifest jar doesn't get deleted immediately since that allows the user to re-run the `java` command from the terminal. For example, IntelliJ displays the full `java -classpath ...` invocation when you run tests.

Reply:
If we are going to generate manifest.jar in server side, we probably need to extend `JvmTestEnvironmentParams` to `manifestJar:String` field

Reply:
Following up from an offline conversation. We agreed that option 3 (generate manifest.jar on the intellij side) is the best way to go forward. The reasoning is that this approach allows intellij to delete the manifest jar after running tests. BSP clients are expected to handle too large classpaths anyways (they should not rely on the server producing a manifest jar).

Reply:
Sounds good. I don't have an opinion on this. However, having an option in bloop to generate manifest jars in the future might prove handy.

One approach that hasn't been considered would be a variation of 2: have a deterministic path within `.bloop/$project` where bloop writes the manifest jar used for the last invocation. Every time a test execution happens, it gets overwritten with the manifest used to run tests. This would make it useful for debugging. The manifest jar is just a bunch of paths so I think it's benign to keep the manifest jar with the dated paths in a fixed file location.

Reply:
I agree that option 2 is desirable. I think it's worth implementing to improve the Bloop cli testing experience (easier to debug) but it's not necessary to unblock the IntelliJ BSP testing experience.